// @testing-library/react-native/build/helpers/getByAPI.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getByAPI = exports.UNSAFE_getByType = exports.UNSAFE_getByProps = exports.UNSAFE_getAllByType = exports.UNSAFE_getAllByProps = void 0;

var _prettyFormat = _interopRequireDefault(require("pretty-format"));

var _errors = require("./errors");

var _byTestId = require("./byTestId");

var _byText = require("./byText");

var _byPlaceholderText = require("./byPlaceholderText");

var _byDisplayValue = require("./byDisplayValue");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const UNSAFE_getByType = instance => function getByTypeFn(type) {
  try {
    return instance.findByType(type);
  } catch (error) {
    throw new _errors.ErrorWithStack((0, _errors.prepareErrorMessage)(error), getByTypeFn);
  }
};

exports.UNSAFE_getByType = UNSAFE_getByType;

const UNSAFE_getByProps = instance => function getByPropsFn(props) {
  try {
    return instance.findByProps(props);
  } catch (error) {
    throw new _errors.ErrorWithStack((0, _errors.prepareErrorMessage)(error), getByPropsFn);
  }
};

exports.UNSAFE_getByProps = UNSAFE_getByProps;

const UNSAFE_getAllByType = instance => function getAllByTypeFn(type) {
  const results = instance.findAllByType(type);

  if (results.length === 0) {
    throw new _errors.ErrorWithStack('No instances found', getAllByTypeFn);
  }

  return results;
};

exports.UNSAFE_getAllByType = UNSAFE_getAllByType;

const UNSAFE_getAllByProps = instance => function getAllByPropsFn(props) {
  const results = instance.findAllByProps(props);

  if (results.length === 0) {
    throw new _errors.ErrorWithStack(`No instances found with props:\n${(0, _prettyFormat.default)(props)}`, getAllByPropsFn);
  }

  return results;
};

exports.UNSAFE_getAllByProps = UNSAFE_getAllByProps;

const getByAPI = instance => ({
  getByText: (0, _byText.getByText)(instance),
  getByPlaceholderText: (0, _byPlaceholderText.getByPlaceholderText)(instance),
  getByDisplayValue: (0, _byDisplayValue.getByDisplayValue)(instance),
  getByTestId: (0, _byTestId.getByTestId)(instance),
  getAllByText: (0, _byText.getAllByText)(instance),
  getAllByPlaceholderText: (0, _byPlaceholderText.getAllByPlaceholderText)(instance),
  getAllByDisplayValue: (0, _byDisplayValue.getAllByDisplayValue)(instance),
  getAllByTestId: (0, _byTestId.getAllByTestId)(instance),
  // Unsafe
  UNSAFE_getByType: UNSAFE_getByType(instance),
  UNSAFE_getAllByType: UNSAFE_getAllByType(instance),
  UNSAFE_getByProps: UNSAFE_getByProps(instance),
  UNSAFE_getAllByProps: UNSAFE_getAllByProps(instance)
});

exports.getByAPI = getByAPI;
